packages/tree-view/src/ux-default-tree-view-configuration.ts

```diff
@@ -0,0 +1,5 @@
+import { UxTreeViewTheme } from './ux-tree-view/ux-tree-view-theme';
+
+export class UxDefaultTreeViewConfiguration {
+  public theme?: UxTreeViewTheme;
```

Comment:
I kinda like this idea to have the theme in the configuration. But giving it more thoughts I'm afraid it leads consumers the wrong way... Just want to highlight something here to discuss the "default style scenario".

Without the `theme` configuration, the original way to apply a default style to a component is through the core `styleEngine.applyTheme()` method. When we use this method without passing an `element` then the styles are applied as default in the `<HEAD/>`.

Now I see two benefits from using `applyTheme()`:

1. I think it has better perf as it only apply the theme once
2. More subtle: in this case when we pass a `theme` to a component, it will only override the default theme. On the contrary, with the default `theme` property on the plugin, the component will either use the default, or use the binded property.

My conclusions:
- I like to have the `theme` property as a configuration as it makes it super easy for consumers to set a default theme
- I'm not so confident with the idea to have a either/or scenario for the default theme. I think that if I set a nice button theme as a default but for some reason I need to change only the color at one place, I would prefer to only pass `theme.bind="{background: 'red'}"` instead of a whole new theme
- Seems to me that perf-wise the `applyTheme()` is better

What do you think ? Should we keep the `theme` configurable option but in the `configure()` we do a `applyTheme()` (as a postTask or something like that) instead of applying the default theme in the component constructor ?

Reply:
It does create ambiguity. I would remove it. But leave properties like dense or variant and pass them to inputs

Reply:
I would even remove it from the config as long as StyleEngine is documented